src/main/java/org/commcare/modern/database/DatabaseHelper.java

```diff
@@ -76,6 +90,26 @@
 
             set = true;
         }
+        if (inverseFieldNames != null) {
```

Comment:
This should never be null, I think, it should be an empty array.

This code in general should be refactored so that `fieldNames` and `inverseFieldNames` are treated equivalently, so if there's a null or empty check on the latter, there should be one on the former (people should be able to run a query with only inverse matches)